source/src/microapp/cs_Microapp.cpp

```diff
@@ -245,16 +265,18 @@ cs_ret_code_t Microapp::storeState(uint8_t index) {
  */
 bool Microapp::canRunApp(uint8_t index) {
 	LOGMicroappVerbose(
-			"enabled=%u checkSumTest=%u memoryUsage=%u bootTest=%u failedFunction=%u",
-			_states[index].enabled,
+			"checkSumTest=%u enabled=%u bootTest=%u memoryUsage=%u didReboot=%u exceededCallDuration=%u failedFunction=%u",
 			_states[index].checksumTest,
+			_states[index].enabled,
+			_states[index].bootTest,
 			_states[index].memoryUsage,
 			_states[index].didReboot,
-			_states[index].bootTest,
+			_states[index].exceededCallDuration,
 			_states[index].failedFunction);
 	return _states[index].enabled && _states[index].checksumTest == MICROAPP_TEST_STATE_PASSED
 		   && _states[index].memoryUsage != 1 && _states[index].bootTest != MICROAPP_TEST_STATE_FAILED
-		   && _states[index].didReboot != true && _states[index].failedFunction == MICROAPP_FUNCTION_NONE;
+		   && _states[index].didReboot != true && _states[index].exceededCallDuration != true
+		   && _states[index].failedFunction == MICROAPP_FUNCTION_NONE;
 }
 
 void Microapp::tick() {
```

Comment:
Is _currentMicroappIndex really a signed it just to be able to set it to -1?

Reply:
yep